[Zwe Zeya] iP

ZweZeya opened the pull request:
# Avo
> "A lot of hard work is hidden behind nice things." &ndash; [Ralph Lauren](https://en.wikipedia.org/wiki/Ralph_Lauren)

Avo frees your mind from having to remember things you need to do. It's

- text-based
- easy to learn
- ~FAST~ *SUPER* FAST to use

All you need to do is

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you are a Java programmer, you can also use it to practice Java. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ZweZeya committed:
Rename chatbot and update greeting and exit messages

ZweZeya committed:
Add echo feature based on user input

ZweZeya committed:
Add storage for user inputs

ZweZeya committed:
Add Task and TaskManager to handle tasks completion

ZweZeya committed:
Add different types of tasks

ZweZeya committed:
Add initial testing and reformat string output

ZweZeya committed:
Restructure file and refactor Avo code

ZweZeya committed:
Add exception handling for invalid inputs

ZweZeya committed:
Support deletion of tasks

ZweZeya committed:
Refactor code to use switch statement

ZweZeya committed:
Refactor command logic to use enums

ZweZeya committed:
Fix scanner last line bug

ZweZeya committed:
Fix scanner last line bug

ZweZeya committed:
Save tasks to file

ZweZeya committed:
Handle dates of tasks

ZweZeya committed:
Refactor code with separation of concerns

ZweZeya committed:
Decouple all components

ZweZeya committed:
Fix storage format bug

ZweZeya committed:
Merge branch 'add-gradle-support'

ZweZeya committed:
Setup gradle

ZweZeya committed:
Refactor code into packages

ZweZeya committed:
Setup JUnit for unit testing

ZweZeya committed:
Change gradle for jar archive

ZweZeya committed:
Add searching by name

ZweZeya committed:
Fix import code style

ZweZeya committed:
Add javadocs for some classes

ZweZeya committed:
Merge branch 'branch-A-JavaDoc'

ZweZeya committed:
Merge branch 'branch-A-CodingStandard'

ZweZeya committed:
Fix code style

ZweZeya committed:
Handle time

ZweZeya committed:
Add tests for TaskParser

ZweZeya committed:
Refactor print statements

ZweZeya committed:
Apply single responsibility for TaskManager

ZweZeya committed:
Fix tests

ZweZeya committed:
Add javadocs

Fui03 commented on src/main/java/seedu/avo/utils/DateTime.java:
```diff
@@ -0,0 +1,41 @@
+            throw new AvoException(generateErrorMessage(format, input));
+        }
+    }
+    public static String format(LocalDateTime date) {
```
**Suggestion**: Could you please consider a more descriptive name for `format` method? It's not immediately clear what is being formatted.

Fui03 commented on src/main/java/seedu/avo/tasks/Task.java:
```diff
@@ -0,0 +1,37 @@
+     public void complete() {
+          isCompleted  = true;
+     }
+     public void unComplete() {
```
**Suggestion**: The `unComplete` method name seems a bit unconventional. Could you please consider a more precise or commonly used term to replace it?

Fui03 commented on src/main/java/seedu/avo/storage/FileStorage.java:
```diff
@@ -0,0 +1,58 @@
+public class FileStorage<T> extends Storage<T, String> {
+    private final String filePath;
+    private final FileParser<T> parser;
+    public FileStorage(String dirPath, FileParser<T> parser) {
```
**Suggestion**: The `dirPath` parameter name is a bit unclear. Could you please think a more descriptive or intuitive name?

Fui03 commented on src/main/java/seedu/avo/parser/CommandParser.java:
```diff
@@ -0,0 +1,24 @@
+     * @return A Command from a store of recognised commands
+     */
+    public Command parse(String input) {
+        String commandStr = input.split(" ")[0];
```
**Suggestion**: The `commandStr` parameter name is a bit unclear. Could you please think a better name for the `commandStr`?

Fui03 reviewed (commented):
Overall, the code is well-written and adheres to the coding standards. Great job! It doesn't violate any major coding practices. However, making the changes I mentioned could further improve clarity. Well done!

ZweZeya committed:
Add description for chatbot

ZweZeya committed:
Add checkstyle

Hoodini231 commented on src/main/java/seedu/avo/commands/Command.java:
```diff
@@ -0,0 +1,11 @@
+ */
+public abstract class Command {
+    public boolean isExit() {
+        return false; }
```
} doesn't follow coding standard me

Hoodini231 commented on src/main/java/seedu/avo/ui/AppUI.java:
```diff
@@ -0,0 +1,118 @@
+        }
+        return input;
+    }
+    private void print(String str) {
```
Potentially confusing method as some use print as their way to displaying information to user and some use System.out.println

this can be confusing from a naming standpoint and the diffs between the two.

Hoodini231 commented on src/test/java/seedu/avo/storage/TaskParserTest.java:
```diff
@@ -0,0 +1,50 @@
+                parser.parse("E : 0 : name : Jan 1 2024 1230 : Jan 2 2024 1230"));
+    }
+    @Test
+    public void testCompletedEvent() throws AvoException {
```
good naming for your test cases !

Hoodini231 commented on src/main/java/seedu/avo/Avo.java:
```diff
@@ -0,0 +1,57 @@
+            run();
+        }
+    }
+    private void listen() throws AvoException {
```
Could add comments here to explain more clearly what listen is doing to someone who may potentially be unaware

Hoodini231 commented on src/main/java/seedu/avo/tasks/Event.java:
```diff
@@ -0,0 +1,46 @@
+        return isOnStartTime || isOnEndTime || isBetweenStartTimeAndEndTime;
+    }
+    @Override
+    public String formatData() {
```
Could be clearer on what this method is formating via javadoc comments

Hoodini231 commented:
The code is solid and follows coding standards—excellent work! There are no significant issues with coding practices.

ZweZeya committed:
Add gui

ZweZeya committed:
Integrate GUI to functionality

ZweZeya committed:
Merge branch 'branch-Level-10'

ZweZeya committed:
Fix style

ZweZeya committed:
Refactor code

ZweZeya committed:
Remove magic numbers

Commands creating the tasks have
magic numbers, hence removing them
improves code readability

ZweZeya committed:
Add assertion to check parsing of file input

Only non-empty lines should be parsed

ZweZeya committed:
Remove more magic strings

More magic strings found in
other commands

ZweZeya committed:
Merge pull request #1 from ZweZeya/branch-A-Assertions

Add assertions to TaskParser

ZweZeya committed:
Merge pull request #2 from ZweZeya/branch-A-CodeQuality

Remove magic numbers from Command

ZweZeya committed:
Add help command

ZweZeya committed:
Add ui image

ZweZeya committed:
Add user guide

ZweZeya committed:
Add exit command

ZweZeya committed:
Change readme and add welcome message

ZweZeya committed:
Update gradle

ZweZeya committed:
Update user guide

ZweZeya committed:
Update user guide

ZweZeya committed:
Make help more user friendly

ZweZeya committed:
Change commands to singleton and update tests

ZweZeya committed:
Update gradle

ZweZeya committed:
Add end date validation for event

ZweZeya committed:
Update gradle

ZweZeya committed:
Update user guide

ZweZeya committed:
Update user guide
